homeassistant/components/xiaomi_miio/config_flow.py

```diff
@@ -320,14 +327,24 @@ async def async_step_connect(self, user_input=None):
 
         # Try to connect to a Xiaomi Device.
         connect_device_class = ConnectXiaomiDevice(self.hass)
-        await connect_device_class.async_connect_device(self.host, self.token)
+        try:
+            await connect_device_class.async_connect_device(self.host, self.token)
+        except ConfigEntryAuthFailed:
```

Comment:
I'd raise a custom exception in this helper that we can direct in the caller to a new exception or handle as needed. The config entry exceptions should preferably not be used for other purposes than during entry setup.

Reply:
actually @frenck https://github.com/home-assistant/core/pull/54696#discussion_r695553298 requested to have the ConfigEntryNotReady/ConfigEntryAuthFailed in the location of the helper class (gateway.py).

But I can make the custom exceptions if you want and move them back to the __init__.py file.

Reply:
Yes, we should keep these exceptions for the purpose they are made.

Reply:
Martin is right here, this exception isn't valid to throw in the config flow.